enumarable_methods.rb

```diff
@@ -0,0 +1,74 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+      return to_enum unless block_given?
+    end
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selection = []
+    my_each { |val| selection.push(val) if yield(val) }
+    selection
+  end
+end
+
+def my_all?(value = nil)
+  result = true
+  if value
+    my_each { |element| result &&= element == value }
+  else
+    my_each { |element| result &&= yield(element) }
+  end
+  result
+end
```

Comment:
- [ ] should have working `my_all?` **(covered by automated tests)** identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains truthy values and false if it contains at least one falsy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_all?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_all?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_all?(5)).`